homeassistant/components/timer/__init__.py

```diff
@@ -232,12 +275,60 @@ def unique_id(self) -> typing.Optional[str]:
 
     async def async_added_to_hass(self):
         """Call when entity is about to be added to Home Assistant."""
-        # If not None, we got an initial value.
-        if self._state is not None:
+        if not self._restore:
+            self._state = STATUS_IDLE
             return
-
+        
+        # Check for previous recorded state
         state = await self.async_get_last_state()
-        self._state = state and state.state == state
+        if state is not None:
```

Comment:
Please use a guard clause. `if state is None: self._state = STATUS_IDLE; return`